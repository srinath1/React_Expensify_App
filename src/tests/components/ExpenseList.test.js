// import React from 'react';
// import { shallow } from 'enzyme';
// import { ExpenseList } from '../../components/ExpenseList';
// import expenses from '../fixtures/expenses'
// test('should render expenselist with expense',()=>{

//     const wrapper=shallow(<ExpenseList expenses={expenses} />)
//     expect(wrapper).toMatchSnapshot()
// })

import React from 'react';
//import ReactShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
//import toJSON from 'enzyme-to-json'
import {ExpenseList} from '../../components/ExpenseList'
import expenses from '../fixtures/expenses'
test('should renderer correctly',()=>{



    // const renderer=new ReactShallowRenderer()
    // renderer.render(<Header />)
    // expect(renderer.getRenderOutput()).toMatchSnapshot();

     const wrapper=shallow(<ExpenseList expenses={expenses} />);
     expect(wrapper).toMatchSnapshot()

    // expect(wrapper.find('h1').text()).toBe('Expensify')

})

test('testing with a empty array',()=>{

const wrapper=shallow(<ExpenseList  expenses={[]} />)
expect(wrapper).toMatchSnapshot()

})