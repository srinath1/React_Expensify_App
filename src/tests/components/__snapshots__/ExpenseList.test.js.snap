// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renderer correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h1>
        Expense List
    </h1>
    <ExpenseListItem
        amount={195}
        createdAt={0}
        description="Gum"
        id="1"
        note=""
    />
    <ExpenseListItem
        amount={109500}
        createdAt={-345600000}
        description="Rent"
        id="2"
        note=""
    />
    <ExpenseListItem
        amount={4500}
        createdAt={345600000}
        description="Credit Card"
        id="3"
        note=""
    />
</div>,
  "nodes": Array [
    <div>
      <h1>
            Expense List
      </h1>
      <ExpenseListItem
            amount={195}
            createdAt={0}
            description="Gum"
            id="1"
            note=""
      />
      <ExpenseListItem
            amount={109500}
            createdAt={-345600000}
            description="Rent"
            id="2"
            note=""
      />
      <ExpenseListItem
            amount={4500}
            createdAt={345600000}
            description="Credit Card"
            id="3"
            note=""
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <ExpenseList
      expenses={
            Array [
                  Object {
                    "amount": 195,
                    "createdAt": 0,
                    "description": "Gum",
                    "id": "1",
                    "note": "",
                  },
                  Object {
                    "amount": 109500,
                    "createdAt": -345600000,
                    "description": "Rent",
                    "id": "2",
                    "note": "",
                  },
                  Object {
                    "amount": 4500,
                    "createdAt": 345600000,
                    "description": "Credit Card",
                    "id": "3",
                    "note": "",
                  },
                ]
      }
/>,
    "_instance": null,
    "_newState": null,
    "_rendered": <div>
      <h1>
            Expense List
      </h1>
      <ExpenseListItem
            amount={195}
            createdAt={0}
            description="Gum"
            id="1"
            note=""
      />
      <ExpenseListItem
            amount={109500}
            createdAt={-345600000}
            description="Rent"
            id="2"
            note=""
      />
      <ExpenseListItem
            amount={4500}
            createdAt={345600000}
            description="Credit Card"
            id="3"
            note=""
      />
</div>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ExpenseList
    expenses={
        Array [
            Object {
              "amount": 195,
              "createdAt": 0,
              "description": "Gum",
              "id": "1",
              "note": "",
            },
            Object {
              "amount": 109500,
              "createdAt": -345600000,
              "description": "Rent",
              "id": "2",
              "note": "",
            },
            Object {
              "amount": 4500,
              "createdAt": 345600000,
              "description": "Credit Card",
              "id": "3",
              "note": "",
            },
          ]
    }
/>,
}
`;

exports[`testing with a empty array 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h1>
        Expense List
    </h1>
    <p>
        No Expenses
    </p>
</div>,
  "nodes": Array [
    <div>
      <h1>
            Expense List
      </h1>
      <p>
            No Expenses
      </p>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <ExpenseList
      expenses={Array []}
/>,
    "_instance": null,
    "_newState": null,
    "_rendered": <div>
      <h1>
            Expense List
      </h1>
      <p>
            No Expenses
      </p>
</div>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ExpenseList
    expenses={Array []}
/>,
}
`;
